src/session_impl.cpp

```diff
@@ -5664,7 +5669,7 @@ namespace {
 
 		if (!s->natpmp_mapper
 			&& !(s->flags & listen_socket_t::local_network)
-			&& !(s->flags & listen_socket_t::proxy))
+			&& m_settings.get_bool(settings_pack::listen_on_proxy))
```

Comment:
if you don't configure a proxy and leave `listen_on_proxy` as `false`; it looks like this change would disable NAT-PMP in that case. I think what you meant to do was to add another condition to the `flags & proxy` check.

Reply:
I think there might be a simpler way of doing this. Rather than checking the settings here, I think you should be able to check the `listen_socket_t::accept_incoming` flag. You'd have to make sure this flag is set correctly depending on the `listen_on_proxy` setting when the sockets are created though

Reply:
@arvidn would it be acceptable to just remove the proxy being a factor in whether the user can enable natpmp? Or is this something that is best left in another PR/Issue/Discussion?